src/accumulator/proof.rs

```diff
@@ -245,7 +251,13 @@ impl Proof {
             },
         ))
     }
-    pub fn create_root_candidates(
+
+    /// This function computes a set of roots from a proof.
+    /// If some target's hashes are null, then it computes the roots after
+    /// those targets are deleted.
+    /// It's the caller's responsibility to null out the targets if desired by
```

Comment:
I actually think this function should not be public, not sure why the API consumers would want such a lol-level function. What do you think?

Reply:
Yeah it'd be better if it wasn't public. My assumption was that it's public because of rust modules stuff. Would prefer it to be private.

Reply:
I made it pub(crate), so it can only be used inside the crate and not by API consumers. We need `calculate_roots` for stump updates, and thus it can't be private.

Reply:
Ah ok. That sounds ok.